pkg/build/controller/config_controller.go

```diff
@@ -79,7 +79,7 @@ func (c *BuildConfigController) HandleBuildConfig(bc *buildapi.BuildConfig) erro
 		if kerrors.IsConflict(err) {
 			instantiateErr = fmt.Errorf("unable to instantiate Build for BuildConfig %s/%s due to a conflicting update: %v", bc.Namespace, bc.Name, err)
 			utilruntime.HandleError(instantiateErr)
-		} else if buildgenerator.IsFatal(err) {
+		} else if buildgenerator.IsFatal(err) || kerrors.IsNotFound(err) || kerrors.IsBadRequest(err) {
```

Comment:
those methods don't return a badrequest, but if they did i would assume so...  if a request is bad, is there a reason to think retrying it will help?

if we're returning "badrequest" on something that isn't actually a bad request, then we're doing it wrong, i think.